regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java

```diff
@@ -31,24 +35,66 @@
 
 public class EmptyAlternativeFinder extends RegexBaseVisitor {
 
-  private static final String MESSAGE = "Remove this empty alternative.";
+  private static final String MESSAGE_REMOVE_THIS = "Remove this empty alternative.";
 
   private final RegexIssueReporter.ElementIssue regexElementIssueReporter;
 
+  private int nestedGroupLevel = 0;
+
   public EmptyAlternativeFinder(RegexIssueReporter.ElementIssue regexElementIssueReporter) {
     this.regexElementIssueReporter = regexElementIssueReporter;
   }
 
+  @Override
+  public void visitCapturingGroup(CapturingGroupTree tree) {
+    nestedGroupLevel++;
+    super.visitCapturingGroup(tree);
+    nestedGroupLevel--;
+  }
+
+  @Override
+  public void visitNonCapturingGroup(NonCapturingGroupTree tree) {
+    nestedGroupLevel++;
+    super.visitNonCapturingGroup(tree);
+    nestedGroupLevel--;
+  }
+
+  @Override
+  public void visitAtomicGroup(AtomicGroupTree tree) {
+    nestedGroupLevel++;
+    super.visitAtomicGroup(tree);
+    nestedGroupLevel--;
+  }
+
+  @Override
+  public void visitLookAround(LookAroundTree tree) {
+    nestedGroupLevel++;
+    super.visitLookAround(tree);
+    nestedGroupLevel--;
+  }
```

Comment:
It's the second time, that we have to visit all sub kinds of GroupTree maybe it's worth creating a visitGroup method in RegexBaseVisitor. WDYT? I would make both Finder implementations more readable.

Reply:
Yes, I had the same though.